AdfsEventsModule.psm1

```diff
@@ -4,6 +4,130 @@
+	gpupdate /force
+
+	#Enable ADFS Tracing log
+	$ADFSTraceLogName = "AD FS Tracing/Debug"
+	$ADFSTraceLog = New-Object System.Diagnostics.Eventing.Reader.EventlogConfiguration $ADFSTraceLogName
+	if($ADFSTraceLog.IsEnabled -ne $true)
+	{
+		$ADFSTraceLog.IsEnabled = $true
+		$ADFSTraceLog.SaveChanges()
+	}
+
+	#Enable security auditing from ADFS
+	switch ($OSVersion.Buildnumber)
+	{
+		'6000'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6001'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6002'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7600'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7601'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'9200'{Import-Module ADFS -ErrorAction SilentlyContinue}
+		'9600'{Import-Module ADFS -ErrorAction SilentlyContinue}
+		'14393'{Import-Module ADFS -ErrorAction SilentlyContinue}
+	}
+
+    $SyncProps = Get-ADFSSyncProperties
+    if ($SyncProps.Role -ne 'SecondaryComputer') 
+    {
+        Set-ADFSProperties -LogLevel  @( "FailureAudits", "SuccessAudits", "Warnings", "Verbose", "Errors", "Information")
+    }
+	auditpol.exe /set /subcategory:"Application Generated" /failure:enable /success:enable
+
+	Write-host "ADFS auditing is now enabled."
+}
+
+#function to enable ADFS auditing. Takes in parameters of server names
+function Disable-ADFSAuditing
+{
+	<#
+    .SYNOPSIS
+    This script disables ADFS verbose related events from the security, admin, and debug logs.
+
+    .DESCRIPTION
+	To track ADFS authentication processing there are multiple items which must be enabled on the ADFS server(s). This function provides 
+    automation for disabling those items so that event logs do not fill up.
+    Note that this function can only run on the local server-it cannot be run against a set of servers.
+	
+	EXAMPLE
+    	Disable-ADFSAuditing
+	#>
+	
+	#Disable ADFS Tracing log
+	$ADFSTraceLogName = "AD FS Tracing/Debug"
+	$ADFSTraceLog = New-Object System.Diagnostics.Eventing.Reader.EventlogConfiguration $ADFSTraceLogName
+	if ($ADFSTraceLog.IsEnabled -ne $false)
+	{
+		$ADFSTraceLog.IsEnabled = $false
+		$ADFSTraceLog.SaveChanges()
+	}
+
+	#Disable security auditing from ADFS
+	$cs = get-wmiobject -class win32_computersystem -ComputerName "localhost"
+	$DomainRole = $cs.domainrole
+	$OSVersion = gwmi win32_operatingsystem
+	[int]$BuildNumber = $OSVersion.BuildNumber 
+	switch ($OSVersion.Buildnumber)
+	{
+		'6000'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6001'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6002'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7600'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7601'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'9200'{Import-Module ADFS -ErrorAction SilentlyContinue}

```

Comment:
This block is the same as above.  Factor into a separate function?

Reply:
Agreed, moving to its own function